services/minespace-web/src/components/layout/NotificationDrawer.tsx

```diff
@@ -177,7 +177,11 @@ const NotificationDrawer: FC<INotificationDrawerProps> = (props) => {
     await handleMarkAsRead(notification.notification_guid);
     handleCollapse();
     const routeSpecifics = navigationHandler(notification);
-    const route = { pathname: routeSpecifics?.route, state: { ...(routeSpecifics?.state ?? {}) } };
+    const route = {
+      pathname: routeSpecifics?.route,
+      search: routeSpecifics?.search,
+      state: { ...(routeSpecifics?.state ?? {}) },
+    };
```

Comment:
This was a really confusing bug!
It blew up my routing when I tried to click on the Notice of Departure notification (which then navigates to the NOD page of the dashboard and opens the relevant modal). It found no match every time I clicked on it, and then it would work when I refreshed the page. 😂 
As it turns out, if you are passing in an _object_ to history.push, it creates problems to concatenate the search to the pathname. I tested, and there is no problem passing in a _string_ (like the one returned from dynamicRoute) with it concatenated, but with the object it seems to get confused parsing it.

Reply:
I wonder if this comment should be a code comment or a git commit comment? PR comments will not be found by future developers.